core/common/src/test/java/alluxio/SetAndRestoreAuthenticatedUser.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import alluxio.security.User;
+import alluxio.security.authentication.AuthenticatedClientUser;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * An AutoCloseable which temporarily modifies authenticated user on server side when it is
+ * constructed and restores the authenticated user when it is closed.
+ */
+@ThreadSafe
+public final class SetAndRestoreAuthenticatedUser implements AutoCloseable {
```

Comment:
@jsimsa  I will create a separate PR to consolidate these rules, leaving this PR small